### PR TITLE
CMake: check for timed pthreads functions

### PR DESCRIPTION
**Pull request by djpohly**
The macros `HAVE_PTHREAD_MUTEX_TIMEDLOCK` and `HAVE_PTHREAD_COND_TIMEDWAIT` are still being used in the Pthreads threading implementation, but the CMake build system never sets them, so the timed versions aren't used.

This ended up making WorkerThread heartbeats not beat. </3

### CODE REVIEW

**Comment by djpohly** [comment]
It may be worthwhile and fairly easy to compare the autotools config.h.in to the CMake config.hpp.in and make sure that any other macros which are missing aren't used anywhere in the codebase.

**Comment by kyzentun** [comment]
Leaving this one for @wolfman2000 to check, since it suggests checking the old autotools config.

**Comment by djpohly** [comment]
Sure thing.  Also, I'm much more familiar with autotools, so someone who knows Cmake should make sure I'm Cmaking right.

File: StepmaniaCore.cmake [hunk 1]
```diff
@@ -231,6 +231,9 @@ find_package(Iconv)
 find_package(Threads)
 if (${Threads_FOUND})
   set(HAS_PTHREAD TRUE)
+  list(APPEND CMAKE_REQUIRED_LIBRARIES pthread)
+  check_symbol_exists(pthread_mutex_timedlock pthread.h HAVE_PTHREAD_MUTEX_TIMEDLOCK)
```
Comment (wolfman2000) [inline StepmaniaCore.cmake]: ...admittedly not the most standard of places to put this, but I understand where you are coming from.

File: StepmaniaCore.cmake [hunk 2]
```diff
@@ -231,6 +231,9 @@ find_package(Iconv)
 find_package(Threads)
 if (${Threads_FOUND})
   set(HAS_PTHREAD TRUE)
+  list(APPEND CMAKE_REQUIRED_LIBRARIES pthread)
```
Comment (wolfman2000) [inline StepmaniaCore.cmake]: The only line that bothers me at a glance is this one. Why do we have to manually append pthread here? Won't checking the symbols be enough?

**Comment by djpohly** [comment]
The library wasn't being used for the CMake checks themselves.  Without this line, the pthread_mutex_timedwait symbol wasn't found even when it exists because it required linking with -lpthread.

**Comment by wolfman2000** [comment]
That was the explanation I was waiting for.

**Pull request merged**